src/TritonCTS/src/SinkClustering.cpp

```diff
@@ -167,6 +168,221 @@ void SinkClustering::findBestMatching()
   }
 }
 
+
+void SinkClustering::runAgglomerative(unsigned groupSize, float maxDiameter, cts::DBU scaleFactor)
+{
+  _scaleFactor = scaleFactor;
+  normalizePoints(maxDiameter);
+  computeAllThetas();
+  sortPoints();
+
+  float THRESHOLD_MAX_POINTS = 25000;
+  int numberPoints = _points.size();
+
+  int initnclusters = ceil((float)numberPoints/THRESHOLD_MAX_POINTS);
+  int maxClusterSize = ceil((float)numberPoints/initnclusters);
+
+  std::vector<std::vector<unsigned>> thetaSolution;
+
+
+  int currentIndex = 0;
+  
+  for(unsigned i=0; i<_thetaIndexVector.size(); ++i){
+
+    if(thetaSolution.size() == 0){
+      std::vector<unsigned> indexesVector;
+      thetaSolution.push_back(indexesVector);
+      currentIndex = 0;
+    }
+    unsigned idx = _thetaIndexVector[i].second;
+    if(thetaSolution[currentIndex].size() < maxClusterSize){
+      thetaSolution[currentIndex].push_back(idx);
+    }
+    else{
+      std::vector<unsigned> indexesVector;
+      thetaSolution.push_back(indexesVector);
+      currentIndex++;
+    }
+  }
+
+  SinkAgglClustering agglClus;
+
+  std::vector<std::vector<unsigned>> clusterSolution;
+  std::vector<unsigned> pointsToCluster;
+
+  for(unsigned clusterCounter=0; clusterCounter<thetaSolution.size(); ++clusterCounter){
+
+    // Empty Vector
+    pointsToCluster.clear();
+
+    // Agglomerative for each cluster..
+    for(unsigned pointCounter=0; pointCounter<thetaSolution[clusterCounter].size(); ++pointCounter){
+      pointsToCluster.push_back(thetaSolution[clusterCounter][pointCounter]);
+    }
+
+    // Keep Running Till All Cluster Size <= GroupSize and Cluster Diameter <= _maxDiameter
+    while(true)
```

Comment:
Is it possible to define the loop condition in the while loop itself rather than relying on breaks to exit the loop. 


while(true) is a pretty scary proposition as well, if there's ever a bug introduced it will hang the entire program. Is there some sanity check we can include that will help escape this loop if it enters an infinite loop.